### ISSUE TITLE
Failed from timeout: Could not create Thumbnail

### ISSUE THREAD

**Issue by bkhezry** [GeoNode/geonode#4517]
## Expected Behavior
Create thumbnail of map

## Actual Behavior
Return with `Failed from timeout: Could not create Thumbnail`

## Steps to Reproduce the Problem

  1. Using Edit Tools 
  2. Press **set** button under Thumbnail

## Specifications

  - GeoNode version: **2.10rc5**
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker): **GeoNode Docker**
  - Platform: **Ubuntu 16.04**

**Comment by bkhezry** [comment]
I just test http://master.demo.geonode.org/maps/80 map for creating thumbnail and it not working.
With wizard option in editing tools, the thumbnail preview of the map did not work too.

![Selection_031](https://user-images.githubusercontent.com/91809/59569961-e192ab00-90a5-11e9-9726-07ad9736ca5c.png)

**Comment by gannebamm** [comment]
Hi @bkhezry,
Do you have some GeoServer logs for us?
And could you check if this PR will fix the timeout: https://github.com/GeoNode/geonode/pull/4509
It is just one line in the settings.py you have to change.

**Comment by gannebamm** [comment]
Btw. I encountered one instance in which gefence was not fetching the correct tiles from webcache and this culminating into the timeout. Releasing the old geofence cache fixed the issue for me (until now). In addition I **lowered** the fence cache invalidation time, hoping this will get rid of old cache soon enough to prevent this.

**Comment by bkhezry** [comment]
Hi @gannebamm,
I change the URL of the base map for a thumbnail to Wikipedia & time out error fixed. but thumbnail didn't create and the white image generates for map.

**Comment by capooti** [comment]
hi @gannebamm 
I confirm the error is still there unfortunately - same error like before. Here is the GeoServer log:

```
2019-06-27 15:17:11,088 INFO [geoserver.flow] - releasing flow controllers for [WMS 1.1.1 GetMap]
2019-06-27 15:17:11,088 INFO [geoserver.flow] - Request completed, running requests: 0, blocked requests: 0
2019-06-27 15:17:11,171 WARN [geoserver.printng] - Error fetching : http://a.tile.openstreetmap.org/5/28/18.png, status is : 403
2019-06-27 15:17:11,336 WARN [geoserver.printng] - Error fetching : http://a.tile.openstreetmap.org/5/27/18.png, status is : 403
2019-06-27 15:17:11,365 WARN [geoserver.printng] - Error fetching : http://a.tile.openstreetmap.org/5/28/17.png, status is : 403
2019-06-27 15:17:11,561 WARN [geoserver.printng] - Error fetching : http://c.tile.openstreetmap.org/5/27/17.png, status is : 403
2019-06-27 15:17:11,569 WARN [geoserver.printng] - Error fetching : http://a.tile.openstreetmap.org/5/28/19.png, status is : 403
```

**Comment by gannebamm** [comment]
I just did the following:
* git clone master
* used spc-geonode docker setup with `docker-compose up --build`
* checked if the geoserver auth button works (it did)
* uploaded a shapefile
>> result:
![grafik](https://user-images.githubusercontent.com/6172324/60326562-3083ee80-998a-11e9-906e-ae02435291af.png)

Which setup of geonode did you use @capooti ?

**Comment by gannebamm** [comment]
Ok **but** If you click the create thumbnail button for the exact same layer it will result in an empty background..
```
nginx_1           | 172.18.0.11 - - [28/Jun/2019:07:52:48 +0000] "POST /api/o/v4/tokeninfo/?access_token=xJhkLo2US4hEx4zj9sUGCgxrgiGPfq HTTP/1.1" 200 273 "-" "Java/1.8.0_212"
django_1          | [pid: 40|app: 0|req: 62/221] 172.18.0.11 () {40 vars in 704 bytes} [Fri Jun 28 07:52:48 2019] POST /api/o/v4/tokeninfo/?access_token=xJhkLo2US4hEx4zj9sUGCgxrgiGPfq => generated 273 bytes in 12 msecs (HTTP/1.1 200) 8 headers in 291 bytes (1 switches on core 0)
nginx_1           | 172.18.0.5 - - [28/Jun/2019:07:53:00 +0000] "POST /geoserver/rest/printng/render.png?width=240&auth=172.29.179.101,super,duper&height=200 HTTP/1.1" 200 1820 "-" "python-requests/2.22.0"
nginx_1           | 172.29.179.97 - - [28/Jun/2019:07:53:00 +0000] "POST /layers/geonode:someHydroData/thumbnail HTTP/1.1" 200 15 "http://172.29.179.101/layers/geonode:someHydroData" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:64.0) Gecko/20100101 Firefox/64.0"
```

**Comment by afabiani** [comment]
The "set thumbnail" button uses the layers rendered on the currently visible map.

In order to fix this, we would need to change the background layers from openlayers to wikimedia.

This is something that must be done through settings

**Comment by capooti** [comment]
yes, correct, layer upload works, but set thumbnail don't and definitely agree with @afabiani

**Comment by afabiani** [comment]
Be aware that there could be some copyright issues.

We need to be sure that:

1. The attribution is correctly displayed on maps

2. The documentation says explicitly that developers are not responsible for using those backgrounds for commercial products based on GeoNode

**Comment by capooti** [comment]
I agree @afabiani 
I would suggest that by default thumbnails are generated without base layers. We could have a setting which enables that, and we should point out that by enabling that setting there could be implication of copyright etc etc
